src/mlpack/methods/ann/layer/alpha_dropout.hpp

```diff
@@ -0,0 +1,183 @@
+   * @param output Resulting output activation.
+   */
+  template<typename eT>
+  void Forward(const arma::Mat<eT>&& input, arma::Mat<eT>&& output);
+
+  /**
+   * Ordinary feed backward pass of the alpha_dropout layer.
+   *
+   * @param input The propagated input activation.
+   * @param gy The backpropagated error.
+   * @param g The calculated gradient.
+   */
+  template<typename eT>
+  void Backward(
+      const arma::Mat<eT>&& /* input */,
+      arma::Mat<eT>&& gy,
+      arma::Mat<eT>&& g);
+
+  //! Get the input parameter.
+  InputDataType const& InputParameter() const { return inputParameter; }
+  //! Modify the input parameter.
+  InputDataType& InputParameter() { return inputParameter; }
+
+  //! Get the output parameter.
+  OutputDataType const& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return outputParameter; }
+
+  //! Get the detla.
+  OutputDataType const& Delta() const { return delta; }
+  //! Modify the delta.
+  OutputDataType& Delta() { return delta; }
+
+  //! The value of the deterministic parameter.
+  bool Deterministic() const { return deterministic; }
+  //! Modify the value of the deterministic parameter.
+  bool& Deterministic() { return deterministic; }
+
+  //! The probability of setting a value to alpha_dash.
+  double Ratio() const { return ratio; }
+
+  //! Value to be multiplied with x for affine transformation.
+  double A() const { return a; }
+
+  //! Value to be added to a*x for affine transformation.
+  double B() const { return b; }
+
+  //! Value of alpha_dash.
+  double Alpha_Dash() const {return alpha_dash; }
+
+  //! Get the mask.
+  OutputDataType const& Mask() const {return mask;}
+
+  //! Modify the probability of setting a value to alpha_dash.  As
+  //! 'a' and 'b' depend on 'ratio', modify them as well.
+  void Ratio(const double r)
+  {
+    ratio = r;
+    a = pow((1 - ratio) * (1 + ratio * pow(alpha_dash, 2)), -0.5);
+    b = -a * alpha_dash * ratio;
+  }
+
+
+  /**
+   * Serialize the layer.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const unsigned int /* version */);
+
+ private:
+  //! Locally-stored delta object.
+  OutputDataType delta;
+
+  //! Locally-stored input parameter object.
+  InputDataType inputParameter;
+
+  //! Locally-stored output parameter object.
+  OutputDataType outputParameter;
+
+  //! Locally-stored mast object.
+  OutputDataType mask;
+
+  //! The probability of setting a value to aplha_dash.
+  double ratio;
+
+  //! If true dropout and scaling is disabled, see notes above.
+  bool deterministic;
+
+  //! Value of alpha for normalized inputs (taken from SELU)
+  const double alpha = 1.6732632423543772848170429916717;
+
+  //! Value of lambda for normalized inputs (taken from SELU)
+  const double lambda = 1.0507009873554804934193349852946;
+
+  //! The low variance value of SELU activation function.
+  double alpha_dash = -alpha*lambda;
```

Comment:
I think you can declare each of these constexprs.

Reply:
Do you mean using #define for these?

Reply:
constexpr; see http://en.cppreference.com/w/cpp/language/constant_expression